File: Samples-Xamarin/Fragments/Dataform/DataFormExpandableGroupsFragment.cs
using System;
using Android.Views;
using Com.Telerik.Android.Common;
using Com.Telerik.Widget.Dataform.Visualization;
using Android.OS;
using Android.Content;
using Android.Widget;

namespace Samples
{
	public class DataFormExpandableGroupsFragment : Android.Support.V4.App.Fragment, ExampleFragment, IFunction2
	{
		public override View OnCreateView (LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
		{
			ViewGroup rootLayout = (ViewGroup)inflater.Inflate(Resource.Layout.fragment_dataform_grouping, null);

			RadDataForm dataForm = new RadDataForm(this.Activity);

			DataFormGroupLayoutManager groupManager = new DataFormGroupLayoutManager(this.Activity);
			groupManager.CreateGroup = this;

			dataForm.LayoutManager = groupManager;
			dataForm.SetEntity(new Person());

			rootLayout.AddView(dataForm);

			return rootLayout;
		}

		public Java.Lang.Object Apply(Java.Lang.Object context, Java.Lang.Object groupName) {
			ExpandableEditorGroup group = new ExpandableEditorGroup((Context)context, groupName.ToString());
			group.AddIsExpandedChangedListener (new MyIsExpandedChangedListener (this.Activity, group.Name ()));
			return group;
		}

		public string Title() {
			return "Data Form Expandable Groups";
		}
	}

	public class MyIsExpandedChangedListener : Java.Lang.Object, Com.Telerik.Widget.Dataform.Visualization.ExpandableEditorGroup.IIsExpandedChangedListener {
		
		Context context;
		String groupName;

		public MyIsExpandedChangedListener(Context context, String groupName) {
			this.context = context;
			this.groupName = groupName;
		}

		public void OnChanged (bool isExpanded)
		{
			String message = String.Format("{0} has been {1}", groupName, isExpanded ? "expanded" : "collapsed");
			Toast.MakeText(context, message, ToastLength.Short).Show();
		}
	}
}

File: Samples-Java/Samples/src/main/java/fragments/dataform/DataFormExpandableGroupsFragment.java
package fragments.dataform;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.telerik.android.common.Function2;
import com.telerik.android.sdk.R;
import com.telerik.widget.dataform.visualization.DataFormGroupLayoutManager;
import com.telerik.widget.dataform.visualization.EditorGroup;
import com.telerik.widget.dataform.visualization.ExpandableEditorGroup;
import com.telerik.widget.dataform.visualization.RadDataForm;

import activities.ExampleFragment;

public class DataFormExpandableGroupsFragment extends Fragment implements ExampleFragment {
    @Override
    public String title() {
        return "Data Form Expandable Groups";
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        ViewGroup rootLayout = (ViewGroup) inflater.inflate(R.layout.fragment_dataform_grouping, container, false);

        RadDataForm dataForm = (RadDataForm)rootLayout.findViewById(R.id.data_form_grouping);

        DataFormGroupLayoutManager groupManager = new DataFormGroupLayoutManager(this.getActivity());
        groupManager.setCreateGroup(new Function2<Context, String, EditorGroup>() {
            @Override
            public EditorGroup apply(Context context, String groupName) {
                final ExpandableEditorGroup group = new ExpandableEditorGroup(context, groupName);
                group.addIsExpandedChangedListener(new ExpandableEditorGroup.IsExpandedChangedListener() {
                    @Override
                    public void onChanged(boolean isExpanded) {
                        String message = String.format("%s has been %s", group.name(), isExpanded ? "expanded" : "collapsed");
                        Toast.makeText(getContext(), message, Toast.LENGTH_SHORT).show();
                    }
                });
                return group;
            }
        });

        dataForm.setLayoutManager(groupManager);
        dataForm.setEntity(new Person());

        return rootLayout;
    }
}
